### PR TITLE
Add validation that referenced project target framework is compatible with current target

### LINKED ISSUE #264: All invalid p2p references should result in warnings/errors
We're moving to a model where builds via the command-line and Visual Studio are exactly the same, no different validation, no different warnings, etc.

The first step of this, is to push P2P validation to MSBuild indirectly via ResolveProjectReferences. The project system will then pick up these warnings/errors during builds and design-time builds and push them to the Error List and [stop blocking you from adding a reference](https://github.com/dotnet/roslyn-project-system/pull/595).

We already do some validating in the multi-TFM case, this bug is representing the work to do similar validation in the single-TFM case, and to make sure that we're erroring/warning in the right cases.

### PR DESCRIPTION
**Pull request by dsplaisted**
This fixes #264 for the case where both projects use the .NET SDK.

### CODE REVIEW

**Comment by dsplaisted** [comment]
@nguerrera @srivatsn

File: test/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+            using (var file = File.CreateText(targetProjectPath))
+            {
+                projectXml.Save(file);
+            }
+        }
+
+        public override string ToString()
+        {
+            StringBuilder ret = new StringBuilder();
```
Comment (nguerrera) [inline test/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs]: nit: use var

File: test/Microsoft.NET.TestFramework/TestAsset.cs [hunk 2]
```diff
@@ -60,7 +84,7 @@ public TestAsset WithSource()
                         .Descendants(XName.Get("PackageReference", "http://schemas.microsoft.com/developer/msbuild/2003"))
                         .FirstOrDefault(pr => pr.Attribute("Include")?.Value == "Microsoft.NET.Sdk")
                         ?.Element(XName.Get("Version", "http://schemas.microsoft.com/developer/msbuild/2003"))
-                        ?.SetValue('[' +_buildVersion + ']');
+                        ?.SetValue('[' +BuildVersion + ']');
```
Comment (nguerrera) [inline test/Microsoft.NET.TestFramework/TestAsset.cs]: nit: whitespace missing between + and BuildVersion. Also, consider $"[{BuildVersion}]".

**Review by nguerrera — APPROVED** [review]

**Comment by nguerrera** [comment]
LGTM, but tests are failing.

**Comment by nguerrera** [comment]
Still failing on Linux. I think the test cases involving full framework need to be filtered.

**Review by srivatsn — APPROVED** [review]

**Pull request merged**